src/browser/provider/index.ts

```diff
@@ -150,20 +148,22 @@ export default class BrowserProvider {
             this.localBrowsersInfo[browserId].windowDescriptor = await browserTools.findWindow(browserId);
     }
 
-    private async _ensureBrowserWindowParameters (browserId: string, allowMultipleWindows: boolean): Promise<void> {
+    private async _ensureBrowserWindowParameters (browserId: string): Promise<void> {
         await this._ensureBrowserWindowDescriptor(browserId);
 
         if (OS.win && !this._getResizeCorrections(browserId))
             await this._calculateResizeCorrections(browserId);
         else if (OS.mac && !this._getMaxScreenSize(browserId))
             await this._calculateMacSizeLimits(browserId);
-
-        if (allowMultipleWindows)
-            await this._calculateWindowId(browserId);
     }
 
     private async _closeLocalBrowser (browserId: string): Promise<void> {
-        await browserTools.close(this._getWindowDescriptor(browserId));
+        if (this.plugin.needCleanUpBrowserInfo)
+            this.plugin.cleanUpBrowserInfo(browserId);
+
+        const windowDescriptor = this._getWindowDescriptor(browserId);
+
+        await browserTools.close(windowDescriptor);
```

Comment:
(optional)
previous version is not ugly, I think we do not need to extract const

Reply:
In my opinion, the nested method calls look ugly.